crates/walrus-sui/src/wallet.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Wallet Context wrapper.
+
+use std::{
+    collections::BTreeSet,
+    path::{Path, PathBuf},
+};
+
+use sui_package_management::LockCommand;
+use sui_sdk::{
+    rpc_types::{SuiObjectData, SuiTransactionBlockResponse},
+    sui_client_config::SuiEnv,
+    wallet_context::WalletContext,
+};
+use sui_types::{
+    base_types::{ObjectID, ObjectRef, SuiAddress},
+    crypto::EmptySignInfo,
+    message_envelope::Envelope,
+    transaction::{SenderSignedData, Transaction, TransactionData},
+};
+
+/// The `Wallet` struct wraps the `WalletContext` from the Sui SDK. This allows us to
+/// reduce the scope of the `WalletContext` to only the methods we need.
+pub struct Wallet {
+    wallet_context: WalletContext,
+}
+
+impl std::fmt::Debug for Wallet {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Wallet").finish()
+    }
+}
+
+/// The error type for the `Wallet` struct. Currently there are no custom errors, so we defer to
+/// anyhow::Error.
+pub type WalletError = anyhow::Error;
+
+impl Wallet {
+    /// Create a new Wallet.
+    pub fn new(wallet_context: WalletContext) -> Self {
+        Self { wallet_context }
+    }
+
+    /// Passes through to the `WalletContext` to get the active address.
+    pub fn active_address(&mut self) -> Result<SuiAddress, WalletError> {
+        self.wallet_context.active_address()
+    }
+
+    /// Passes through to the `WalletContext` to sign a transaction.
+    pub fn sign_transaction(&self, transaction_data: &TransactionData) -> Transaction {
+        self.wallet_context.sign_transaction(transaction_data)
+    }
+
+    // TODO: move callsites to the RetriableSuiClient.
```

Comment:
Nit: Can you create a follow-up issue for this (unless you address it already in this PR)?

Reply:
Done.